core/src/main/java/org/jobrunr/jobs/states/FailedState.java

```diff
@@ -40,6 +40,17 @@ public FailedState(String message, Exception exception) {
         this.doNotRetry = isProblematicAndDoNotRetry(exception);
     }
 
+    public FailedState(String message, String exceptionType, String exceptionMessage, String exceptionCauseType, String exceptionCauseMessage, String stackTrace, boolean doNotRetry) {
+        super(StateName.FAILED);
+        this.message = message;
+        this.exceptionType = exceptionType;
+        this.exceptionMessage = exceptionMessage;
+        this.exceptionCauseType = exceptionCauseType;
+        this.exceptionCauseMessage = exceptionCauseMessage;
+        this.stackTrace = stackTrace;
+        this.doNotRetry = doNotRetry;
+    }
```

Comment:
The other available "non-no-arg-constructor" eats an `Exception`. I also tried making the restore function `public static` but figured it's more reasonable to skip the restore-and-extract step entirely.

Reply:
Looks good to me! I would only make sure the other constructors use this one.

Reply:
While we're at it, it may be worth caching the reconstructed exception built in the `getException` method by updating the `exception` field. Is there any reason why this isn't done already?